src/domain-functions.ts

```diff
@@ -168,7 +168,7 @@ function pipe<T extends DomainFunction[]>(
   ...fns: T
 ): DomainFunction<Last<UnpackAll<T>>> {
   const last = <T>(ls: T[]): T => ls[ls.length - 1]
-  return map(sequence(...fns), last)
+  return map(sequence(...fns), last) as DomainFunction<Last<UnpackAll<T>>> 
```

Comment:
What happened here?

Reply:
this is unrelated but the tests would break in the latest deno version without this change.
They have [upgraded TS to 5.4.3](https://deno.com/blog/v1.42) and it changed the way TS resolves the type here.

After the upgrade I get
```
error: TS2322 [ERROR]: Type 'DomainFunction<unknown>' is not assignable to type 'DomainFunction<Last<UnpackAll<T>>>'.
  Type 'unknown' is not assignable to type 'Last<UnpackAll<T>>'.
  return map(sequence(...fns), last)
  ~~~~~~
    at file:///home/diogo/Projects/seasoned/domain-functions/src/domain-functions.ts:171:3
```

It resolves `map(sequence(...fns), last)` to `DomainFunction<unknown>`, which is true since we are applying `fns` which is something that extends `DomainFunction<unknown>`. Now that I think about it, it surprises me that it was type-checking before.

I can extract this to a separate PR if you think it's too bad to leave it here.